src/helpers/GetDonationDate.ts

```diff
@@ -1,9 +1,28 @@
-import { ethers } from "ethers";
+import { BigNumber, ethers } from "ethers";
 
 import { addresses } from "../constants";
 import { IBaseAddressRecipientAsyncThunk } from "../slices/interfaces";
 
-export const GetDonationDate = async ({ address, recipient, networkID, provider }: IBaseAddressRecipientAsyncThunk) => {
+/**
+ * Gets the date of the first time a user donated to a specific recipient
+ * @param address Current user's address
+ * @param recipient The donation target of the current user
+ * @param networkID ID number of the network the user is currently connected to
+ * @param provider Network provider object
+ * @returns String representation of the date as Month Abbreviation DD, YYYY
+ *          i.e. Feb 13, 2022
+ */
+export const GetFirstDonationDate = async ({
```

Comment:
Why is this using the older `*thunk` style? If all of the fetching is being moved to react-query, shouldn't this be architected in a way to support that?

Reply:
just because this branch was started a month ago before we started on switching Give to react query and I wasn't sure which would be done first/which the priority would be in getting out first. i.e. didn't want everything to be ready for this but be held up by react query changes